src/main/java/com/ase/restservice/ApplicationSecurity.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletResponse;
+
+
+@Configuration
+public class ApplicationSecurity {
+
+  @Resource
+  private UserDetailsService userDetailsService;
+
+  @Autowired
+  private AccountRepository accountRepo;
+  @Autowired
+  private JwtTokenFilter jwtTokenFilter;
+
+  /**
+   * This method checks whether the account asked for
+   * by the client is authorized to the client or not.
+   * @param authentication authenticated client object
+   * @param accountId accountId that client wants to open
+   * @return whether client is allowed to access account or not
+   */
+  public boolean checkAccountId(Authentication authentication, String accountId) {
+    // need to check which client this account has
+    String clientName = authentication.getName();
+    Account account =  accountRepo.findAccountByAccountId(accountId).orElseThrow(() ->
+            new UsernameNotFoundException("Account Not Found with username: " + accountId));
+
+    return clientName.equals(account.getClientId());
+  }
+  public static String getUsernameOfClientLogged() {
+    Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    if (principal instanceof UserDetails) {
+      return ((UserDetails) principal).getUsername();
+    } else {
+      return principal.toString();
+    }
+  }
+
+  /**
+   * This method checks whether a uri request should be granted based on authentication.
+   * @param http the http request
+   * @return allows http to be build to display
+   * @throws Exception unauthorized
+   */
+  @Bean
+  public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+    String[] accountIdWhiteList = {
+            "/accounts/{accountId}/**",
+            "/assets/{accountId}/**"
+    };
+    http.csrf().disable();
+//    http.authorizeRequests().anyRequest().permitAll();
+    http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+
+    http.authorizeRequests()
+            .antMatchers("/auth/login")
+            .permitAll()
+            .antMatchers(accountIdWhiteList)
+            .access("@applicationSecurity.checkAccountId(authentication,#accountId)")
+            .anyRequest().authenticated();
```

Comment:
I think this is the part where 403 Forbidden error is returned if the client doesn't own the resource it's trying to accessing, right? I was wondering if there is a way to customize the error message. When I test with Postman, correct error codes are returned but the response messages are empty.